build/scripts/build.ps1

```diff
@@ -375,14 +501,7 @@ try {
 
     Process-Arguments
 
-    if ($msbuildDir -eq "") {
-        $msbuild = Ensure-MSBuild
-        $msbuildDir = Split-Path -parent $msbuild
-    }
-    else {
-        $msbuild = Join-Path $msbuildDir "msbuild.exe"
-    }
-
+    $msbuild, $msbuildDir = Ensure-MSBuildAndDir -msbuildDir $msbuildDir
```

Comment:
In this case both. The return of `Ensure-MSBUildAndDir` is a 2 element array and Powershell will splat assign to the variables. The real trivia is what happens when the array is length 1 ...